Character.cpp

```diff
@@ -92,26 +92,38 @@ void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
+        if (hitPoints < *initialHitPoints) {hitPoints = *initialHitPoints;}
+        hitPoints *= 1.1;
+        *initialHitPoints = hitPoints;
+
+        if (armor < *initialArmorLevel) {armor = *initialArmorLevel;}
+        armor *= 1.1;
+        *initialArmorLevel = armor;
+
+        if (attackDamage < *initialAttackDamage) {attackDamage = *initialAttackDamage;}
+        attackDamage *= 1.1;
+        *initialAttackDamage = attackDamage;
+
```

Comment:
You're doing the same thing with pairs of variables.
D.R.Y.
write a free helper function and pass your pairs of variables to it
- [ ] revise